_code-samples/modular-tutorials/account-support.js

```diff
@@ -0,0 +1,158 @@
+// ************ Gather Account Info ********************
+// *****************************************************
+
+function gatherAccountInfo() {
+  let accountData = account1name.value + "\n" + account1address.value + "\n" + account1seed.value + "\n"
+  accountData += account2name.value + "\n" + account2address.value + "\n" + account2seed.value
+  resultField.value = accountData
+}
+
+// *****************************************************
+// ********** Distribute Account Info ******************
+// *****************************************************
+
+function distributeAccountInfo() {
+  let accountInfo = resultField.value.split("\n")
+  account1name.value = accountInfo[0]
+  account1address.value = accountInfo[1]
+  account1seed.value = accountInfo[2]
+  account2name.value = accountInfo[3]
+  account2address.value = accountInfo[4]
+  account2seed.value = accountInfo[5]
+}
+
+// *****************************************************
+// ************ Populate Active Form 1 *****************
+// *****************************************************
+
+function populate1() {
+  accountNameField.value = account1name.value
+  accountAddressField.value = account1address.value
+  accountSeedField.value = account1seed.value
+  getXrpBalance()
+}
+
+// *****************************************************
+// ************ Populate Active Form 2 *****************
+// *****************************************************
+
+function populate2() {
+  accountNameField.value = account2name.value
+  accountAddressField.value = account2address.value
+  accountSeedField.value = account2seed.value
+  getXrpBalance()
+}
+
+// *******************************************************
+// **************** Get Xrp Balance  *********************
+// *******************************************************
+
+async function getXrpBalance() {
+  const net = getNet()
+  const client = new xrpl.Client(net)
+  await client.connect()
+  xrpBalanceField.value = await client.getXrpBalance(accountAddressField.value)
+  client.disconnect()
+} // End of getXrpBalance()
+
+// *******************************************************
+// ************** Get Token Balance  *********************
+// *******************************************************
+
+async function getTokenBalance() {
+  let net = getNet()
+  const client = new xrpl.Client(net)
+  results = 'Connecting to ' + getNet() + '....'
+  resultField.value = results
+  await client.connect()   
+  results += '\nConnected.'
+  resultField.value = results
+  const wallet = xrpl.Wallet.fromSeed(accountSeedField.value)
+  results= "\nGetting account balance...\n"
+  const balance = await client.request({
+    command: "gateway_balances",
+    account: wallet.address,
+    ledger_index: "validated",
+  })
+  results += JSON.stringify(balance.result, null, 2)
+  resultField.value = results
+  xrpBalanceField.value = (await client.getXrpBalance(wallet.address))
+  client.disconnect()
+} // End of getTokenBalance()
+
```

Comment:
```suggestion
// *******************************************************
// **************** Scroll Results ***********************
// *******************************************************
let results

async function updateResults() {
  resultField.value += results;
  resultField.scrollTop = resultField.scrollHeight;
}

// End of updateResults()
```

This creates a function to help accommodate adding additional logs and scrolling the results log down to the latest. If implemented, later code would look something like this:
```
  // Prepare transaction
  const transaction = {
    "TransactionType": "CredentialCreate",
    "Account": wallet.address,
    "Subject": subject,
    "CredentialType": credentialType
  }
  
  results = `\n\n===Preparing and Sending Transaction===\n\n${JSON.stringify(transaction, null, 2)}`
  updateResults()
  ```

Reply:
This is a bridge too far at the moment. We can discuss this as a future enhancement.